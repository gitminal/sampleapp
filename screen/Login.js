import { View, Text, StyleSheet, TextInput, StatusBar, Image, ScrollView, useColorScheme } from 'react-native'
import React from 'react'
import { useState } from "react";
import Colors from '../constant/Colors';
import { TouchableOpacity } from 'react-native-gesture-handler';
import Icon from 'react-native-vector-icons/Ionicons';
import constantElement from '../constant/constantElement';
const Login = ({ navigation }) => {
    const [email, setEmail] = useState('');
    const [chEmail, setchEmail] = useState(true);
    const [emailValidError, setEmailValidError] = useState('');

    const [Password, setPassword] = useState('');
    const [chPassword, setchPassword] = useState(true);
    const [passwordValidError, setpasswordValidError] = useState('');
    const handleValidEmail = () => {
        let reg = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w\w+)+$/;
        if (email == "" || email == undefined || email == null) {
            setEmailValidError('Email address must be enter');
            setchEmail(false);
            return false;
        } else if (!reg.test(email)) {
            setEmailValidError('Enter valid email address');
            setchEmail(false);
            return false;
        } else {
            setEmailValidError("");
            setchEmail(true);
            return true;
        }
    };
    const handlePassword = () => {
        var reg = /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$/;
        if (Password == "" || Password == undefined || Password == null) {
            setpasswordValidError('Password  must be enter');
            setchPassword(false);
            return false;
        } else if (!reg.test(Password)) {
            setpasswordValidError('Enter valid Password must contain atleast 8 character, capital & lowercase letter,special & numeric character');
            setchPassword(false);
            return false;
        } else {
            setpasswordValidError("");
            setchPassword(true);
            return true;
        }
    };
    return (
        <ScrollView showsVerticalScrollIndicator={false}>
            <View style={styles.container}>
                <StatusBar barStyle="dark-content" backgroundColor="#ffff" />
                <View style={styles.logintext}>
                    <Text style={styles.textlog}>Login</Text>
                    <Text style={styles.subtext}>Welcome Home My Friend</Text>
                </View>
                <View style={{ marginTop: 12 }}    >
                    <View style={styles.inputView}>
                        <Icon name="mail-outline" size={constantElement.size} style={styles.inputicon} />
                        <TextInput
                            style={styles.TextInput}
                            placeholder="Email"
                            autoCorrect={false}
                            autoCapitalize="none"
                            onChangeText={setEmail}
                            onEndEditing={handleValidEmail}
                            testID="Emailtest"
                        />

                    </View>
                    <View>
                        {chEmail == true ? null : <Text style={{ color: "red" }} testID="error">{emailValidError}</Text>}
                    </View>
                </View>
                <View style={styles.inputView}>
                    <Icon name="lock-closed-outline" size={constantElement.size} style={styles.inputicon} />
                    <TextInput
                        style={styles.TextInput}
                        placeholder="Password"
                        secureTextEntry={true}
                        onChangeText={setPassword}
                        onEndEditing={handlePassword}
                        testID="Passwordtest"

                    />
                </View>
                <View>
                    {chPassword == true ? null : <Text style={{color: Colors.red}}>{passwordValidError}</Text>}
                </View>
                <TouchableOpacity style={styles.loginBtn} onPress={() => {
                    if (handleValidEmail() && handlePassword()) {
                        navigation.replace("Signup")
                    }
                }}>
                    <Text style={styles.loginbtntext} testID="Signuptest">LOG IN</Text>
                </TouchableOpacity>

                <View style={{ marginTop: 20, flexDirection: "row" }}>
                    <View style={styles.remember} />

                    <Text style={styles.textremem}>Remember Pasword</Text>

                </View>
                <Text onPress={() => { navigation.navigate("Forgotpassword") }} testID="Forgotpassword" style={styles.textforgot}>Forgot Pasword ?</Text>
                <Text style={styles.account}>Dont have an account ?</Text>
                <View style={{ flexDirection: "row", alignItems: "center", justifyContent: "center" }}>
                    <Text style={styles.textplease}>Please</Text>
                    <Text onPress={() => { navigation.navigate("Signup") }} testID="Signup" style={styles.textsignup}> SIGN UP</Text>
                </View>
                <View>
                    <View style={{ ...styles.loginfacebook, backgroundColor: Colors.loginfacebookbackground }}>
                        <Image style={styles.image} source={require('../asset/facebook.png')} />
                        <Text style={{ ...styles.loginbtntext, color: Colors.textlogfacebook }}>LOG IN WITH FACEBOOK</Text>
                    </View>
                </View>
                <View>
                    <View style={{ ...styles.loginfacebook, backgroundColor: Colors.loggooglebackground }}>
                        <Image style={styles.image} source={require('../asset/google.png')} />
                        <Text style={{ ...styles.loginbtntext, color: "#ef3038" }}>LOG IN WITH GOOGLE</Text>
                    </View>
                </View>
            </View>
        </ScrollView>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "#fff",
        padding: 30
    },
    logintext: {
        marginTop: 30
    },
    textlog: {
        fontSize: 30,
        color: "black",
        marginTop: 20,
        fontWeight: "bold"
    },
    subtext: {
        fontSize: 14,
        marginTop: 20
    },
    inputView: {
        marginTop: 20,
        flexDirection: "row"
    },
    TextInput: {
        paddingLeft: 50,
        borderBottomWidth: 0.3,
        flex: 1,
        fontSize: 12
    },
    inputicon: {
        marginTop: 15,
        position: "absolute",
        
    },
    loginBtn:
    {
        borderRadius: 5,
        height: 45,
        alignItems: "center",
        justifyContent: "center",
        marginTop: 30,
        backgroundColor: Colors.primaryColor,
    },
    loginbtntext: {
        color: "white",
        fontSize: 18,
        fontWeight: "bold"
    },
    remember: {

        width: 12,
        height: 12,
        marginTop: 4,
        borderColor: Colors.light,
        borderWidth: 0.5,
    },
    textremem: {
        paddingLeft: 15,
        fontSize: 12
    },
    textforgot: {
        width: 100,
        fontSize: 12,
        marginTop: 10,
    },
    account: {
        marginTop: 20,
        textAlign: 'center',
        color: "#404040",
        fontWeight: "400"
    },
    textplease: {
        color: "#404040",
        fontWeight: "400"
    },
    textsignup: {
        color: Colors.primaryColor,
        fontWeight: "500"
    },
    loginfacebook: {
        borderRadius: 5,
        height: 45,
        alignItems: "center",
        marginTop: 10,
        flexDirection: "row",
    },
    image: {
        alignItems: 'flex-start',
        width: 20,
        height: 20,
        margin: 20

    }
})
export default Login;

