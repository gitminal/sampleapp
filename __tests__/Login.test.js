import React from "react";
import { render } from "@testing-library/react-native"
import Login,{validate} from "../screen/Login";
import { shallow, mount } from 'enzyme';
import Adapter from "enzyme-adapter-react-16"
import enzyme from 'enzyme';
import { fireEvent, screen } from "react-native-testing-library";
import renderer from 'react-test-renderer';
import { debug } from "react-native-reanimated";
import { View, Text, TouchableOpacity,styles ,TextInput} from 'react-native';
enzyme.configure({ adapter: new Adapter() });

// let wrapper;
// const setState = jest.fn();
// const useStateSpy = jest.spyOn(React, "useState")
// useStateSpy.mockImplementation((init) => [init, setState]);

// beforeEach(() => {
//     wrapper = enzyme.mount(enzyme.shallow(<Login />).get(0))
// });

// afterEach(() => {
//     jest.clearAllMocks();
// });
describe('<Login />', () => {
    test('renders correctly', () => {
        const rendered = renderer.create(<Login />).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('render default element', () => {
        const { getAllByText, getByPlaceholderText } = render(<Login />)
        expect(getAllByText("Login").length).toBe(1);
        expect(getAllByText("Welcome Home My Friend").length).toBe(1);
        expect(getByPlaceholderText("Email"));
        expect(getByPlaceholderText("Password"));
    })
    it('show Invalid Username and password', () => {
        const { getByTestId, getByText, queryAllByText, getAllByText } = render(<Login />)
        fireEvent.changeText(getByTestId("Emailtest"));
        fireEvent.changeText(getByTestId("Passwordtest"));
        queryAllByText('email address must be enter');
        expect(queryAllByText('Password  must be enter').length).toMatchSnapshot();
    })
  
    it('go to signup', () => {
        const navigation = { navigate: () => { } }
        spyOn(navigation, 'navigate');
        const page = render(<Login navigation={navigation} />)
        const Signup = page.getByTestId('Signup');
        fireEvent.press(Signup);
        expect(navigation.navigate).toHaveBeenCalledWith("Signup");
    })
    it('go to Forgot', () => {
        const navigation = { navigate: () => { } }
        spyOn(navigation, 'navigate');
        const page = render(<Login navigation={navigation} />)
        const Forgotpassword = page.getByTestId('Forgotpassword');
        fireEvent.press(Forgotpassword);
        expect(navigation.navigate).toHaveBeenCalledWith("Forgotpassword");

    })
  
   
      const container = shallow(<Login />);

      it('should set the password value on change event with trim', () => {
        container.find('TextInput[placeholder="Email"]').simulate('change', {
          target: {
            onchageText: 'Email',
          },
        });
        console.log("container",container);
        expect(container.find('TextInput[placeholder="Email"]').prop('onchageText')).toEqual(
          undefined,
        );
      });
    
     
      const handleValidEmail = jest.fn();
      const END_EDITING = 'content';
      
      const { getByPlaceholderText } = render(
        <View>
          <TextInput placeholder="Email" onEndEditing={handleValidEmail} />
        </View>
      );
      fireEvent.changeText(getByPlaceholderText('Email'),END_EDITING);
      
      test('it changes the value in <TextInput />', () => {
        const handleValidEmail = jest.fn()
        const { queryAllByTestId } = render(<Login onEndEditing={handleValidEmail} />)
        const newTestValue = 'Enter email'
        const input = queryAllByTestId('Emailtest')
        fireEvent.changeText(input, newTestValue) 
        expect(input.handleValidEmail).toHaveBeenCalled(1)
      })
      
      test('it has a placeholder prop', () => {
        const handleValidEmail = jest.fn()
        const { queryAllByTestId } = render(
          <Login onChangeText={handleValidEmail} placeholder="Email" />,
        )
        const input = queryAllByTestId('Emailtest')
        expect(input).toHaveProp('placeholder')
      })
      test('pass', () => {
        const { getByPlaceholderText, queryAllByText } = render(
          <Login />
        );
        fireEvent.changeText(
          getByPlaceholderText('Email'),
          'pen@gmail.com'        );
        const bananaElements = queryAllByText('pen@gmail.com');
        expect(bananaElements).toHaveLength('bananaElements'); 
        console.log(bananaElements);
      });



});