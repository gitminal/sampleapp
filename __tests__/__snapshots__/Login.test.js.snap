// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`<Login /> Title input Should capture title correctly onChange 1`] = `[MockFunction]`;

exports[`<Login /> given empty GroceryShoppingList, user can add an item to it 1`] = `Array []`;

exports[`<Login /> increment counter correctlry 1`] = `[MockFunction]`;

exports[`<Login /> renders correctly 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "padding": 30,
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": 30,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 30,
              "fontWeight": "bold",
              "marginTop": 20,
            }
          }
        >
          Login
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
              "marginTop": 20,
            }
          }
        >
          Welcome Home My Friend
        </Text>
      </View>
      <View
        style={
          Object {
            "marginTop": 12,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginTop": 20,
            }
          }
        >
          <Icon
            name="mail-outline"
            size={16}
            style={
              Object {
                "marginTop": 15,
                "position": "absolute",
              }
            }
          />
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            onChangeText={[Function]}
            onEndEditing={[Function]}
            placeholder="Email"
            style={
              Object {
                "borderBottomWidth": 0.3,
                "flex": 1,
                "fontSize": 12,
                "paddingLeft": 50,
              }
            }
            testID="Emailtest"
          />
        </View>
        <View />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 20,
          }
        }
      >
        <Icon
          name="lock-closed-outline"
          size={16}
          style={
            Object {
              "marginTop": 15,
              "position": "absolute",
            }
          }
        />
        <TextInput
          onChangeText={[Function]}
          onEndEditing={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            Object {
              "borderBottomWidth": 0.3,
              "flex": 1,
              "fontSize": 12,
              "paddingLeft": 50,
            }
          }
          testID="Passwordtest"
        />
      </View>
      <View />
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 20,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#D5D8DC ",
              "borderWidth": 0.5,
              "height": 12,
              "marginTop": 4,
              "width": 12,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 12,
              "paddingLeft": 15,
            }
          }
        >
          Remember Pasword
        </Text>
      </View>
      <Text
        onPress={[Function]}
        style={
          Object {
            "fontSize": 12,
            "marginTop": 10,
            "width": 100,
          }
        }
        testID="Forgotpassword"
      >
        Forgot Pasword ?
      </Text>
      <Text
        style={
          Object {
            "color": "#404040",
            "fontWeight": "400",
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        Dont have an account ?
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#404040",
              "fontWeight": "400",
            }
          }
        >
          Please
        </Text>
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "#7F00FF",
              "fontWeight": "500",
            }
          }
          testID="Signup"
        >
           SIGN UP
        </Text>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ccccff",
              "borderRadius": 5,
              "flexDirection": "row",
              "height": 45,
              "marginTop": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "alignItems": "flex-start",
                "height": 20,
                "margin": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#4169e1",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            LOG IN WITH FACEBOOK
          </Text>
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fbceb1",
              "borderRadius": 5,
              "flexDirection": "row",
              "height": 45,
              "marginTop": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "alignItems": "flex-start",
                "height": 20,
                "margin": 20,
                "width": 20,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#ef3038",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            LOG IN WITH GOOGLE
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<Login /> should match the snapshot 1`] = `"<RCTScrollView><View><View style=\\"flex:1;background-color:#fff;padding:30px\\"><View style=\\"margin-top:30px\\"><Text style=\\"font-size:30px;color:black;margin-top:20px;font-weight:bold\\">Login</Text><Text style=\\"font-size:14px;margin-top:20px\\">Welcome Home My Friend</Text></View><View style=\\"margin-top:12px\\"><View style=\\"margin-top:20px;flex-direction:row\\"><Icon name=\\"mail-outline\\" size=\\"16\\" style=\\"margin-top:15px;position:absolute\\"></Icon><TextInput style=\\"padding-left:50px;border-bottom-width:0.3px;flex:1;font-size:12px\\" placeholder=\\"Email.\\" autoCapitalize=\\"none\\"></TextInput></View><View></View></View><View style=\\"margin-top:20px;flex-direction:row\\"><Icon name=\\"lock-closed-outline\\" size=\\"16\\" style=\\"margin-top:15px;position:absolute\\"></Icon><TextInput style=\\"padding-left:50px;border-bottom-width:0.3px;flex:1;font-size:12px\\" placeholder=\\"Password.\\"></TextInput></View><View></View><View style=\\"margin-top:20px;flex-direction:row\\"><View style=\\"width:12px;height:12px;margin-top:4px;border-color:#D5D8DC;border-width:0.5px\\"></View><Text style=\\"padding-left:15px;font-size:12px\\">Remember Pasword</Text></View><Text style=\\"width:100px;font-size:12px;margin-top:10px\\">Forgot Pasword ?</Text><Text style=\\"margin-top:20px;text-align:center;color:#404040;font-weight:400\\">Dont have an account ?</Text><View style=\\"flex-direction:row;align-items:center;justify-content:center\\"><Text style=\\"color:#404040;font-weight:400\\">Please</Text><Text style=\\"color:#7F00FF;font-weight:500\\"> SIGN UP</Text></View><View><View style=\\"border-radius:5px;height:45px;align-items:center;margin-top:10px;flex-direction:row;background-color:#ccccff\\"><Image style=\\"align-items:flex-start;width:20px;height:20px;margin:20px\\" source=\\"1\\"></Image><Text style=\\"color:#4169e1;font-size:18px;font-weight:bold\\">LOG IN WITH FACEBOOK</Text></View></View><View><View style=\\"border-radius:5px;height:45px;align-items:center;margin-top:10px;flex-direction:row;background-color:#fbceb1\\"><Image style=\\"align-items:flex-start;width:20px;height:20px;margin:20px\\" source=\\"1\\"></Image><Text style=\\"color:#ef3038;font-size:18px;font-weight:bold\\">LOG IN WITH GOOGLE</Text></View></View></View></View></RCTScrollView>"`;

exports[`<Login /> should set the password value on change event with trim 1`] = `undefined`;

exports[`<Login /> show Invalid Username and password 1`] = `0`;
